spec/unit/parser_spec.rb

```diff
@@ -0,0 +1,255 @@
+require 'spec_helper'
+require 'iknow_params/parser'
+
+RSpec.describe IknowParams::Parser do
+  class Controller
+    attr_reader :params
+    def initialize(params)
+      @params = params
+    end
+  end
+
+  before { Controller.include IknowParams::Parser }
+
+  describe "#parse_param" do
+
+    let(:parsed) { Controller.new(@params).parse_param(@param, with: serializer) }
+
+    context "no serializer specified" do
+      let(:serializer) { nil }
+
+      it "returns the value" do
+        @params = {name: "foo"}
+        @param  = :name
+
+        expect(parsed).to eq @params[@param]
+      end
+
+      it "requires the value" do
+        @params = {id: 5}
```

Comment:
You could clean up some of the repetition of structure with a top level `let(:params) { { param => param_value } }`